adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+                      inConnection: Option[Connection] = None,
+                      inAdmin: Option[Admin] = None,
+                      inHBaseContext: Option[HBaseContext] = None) {
+
+    val conf = if (inConf.isDefined) inConf.get else HBaseConfiguration.create()
+    val connection = if (inConnection.isDefined) inConnection.get else ConnectionFactory.createConnection(conf)
+    val admin = if (inAdmin.isDefined) inAdmin.get else connection.getAdmin
+    val hbaseContext = if (inHBaseContext.isDefined) inHBaseContext.get else new HBaseContext(sc, conf)
+
+    def createTable(tableDescriptorMeta: HTableDescriptor) { admin.createTable(tableDescriptorMeta) }
+    def getTable(tableName: TableName): Table = { connection.getTable(tableName) }
+
+    def getHBaseRDD(tableName: org.apache.hadoop.hbase.TableName,
+                    scans: org.apache.hadoop.hbase.client.Scan): RDD[scala.Tuple2[org.apache.hadoop.hbase.io.ImmutableBytesWritable, org.apache.hadoop.hbase.client.Result]] = {
+      hbaseContext.hbaseRDD(tableName, scans)
+
+    }
+
+    def hbaseBulkLoad(genodata: RDD[(Array[Byte], List[(String, Array[Byte])])],
+                      hbaseTableName: String,
+                      flatMap: scala.Function1[(Array[Byte], List[(String, Array[Byte])]), scala.Iterator[scala.Tuple2[org.apache.hadoop.hbase.spark.KeyFamilyQualifier, scala.Array[scala.Byte]]]],
+                      stagingFolder: String) = {
+
+      val familyHBaseWriterOptions = new java.util.HashMap[Array[Byte], FamilyHFileWriteOptions]
+      val f1Options = new FamilyHFileWriteOptions("GZ", "ROW", 128, "FAST_DIFF")
+      familyHBaseWriterOptions.put(Bytes.toBytes("g"), f1Options)
+
+      // call implicit RDD function hbaseBulkLoad to write bulk files to staging locations
+      // and then load to HBase
+      genodata.hbaseBulkLoad(hbaseContext,
+        TableName.valueOf(hbaseTableName),
+        flatMap,
+        stagingFolder,
+        familyHBaseWriterOptions,
+        compactionExclude = false,
+        HConstants.DEFAULT_MAX_FILE_SIZE)
+
+      // This permission change appears necessary, plan to revisit to find better way
+      ("hadoop fs -chmod -R 660 " + stagingFolder) !
+
+      val load = new LoadIncrementalHFiles(conf)
+      load.doBulkLoad(new Path(stagingFolder), admin, connection.getTable(TableName.valueOf(hbaseTableName)),
+        connection.getRegionLocator(TableName.valueOf(hbaseTableName)))
+
+    }
+
+    def hbaseBulkDelete() {}
+
+  }
+
+  /**
+   * KeyStrategy abstract class
+   *
+   * @tparam T Type of rowKeyInfo class which is container for data needed to construct a row key
+   * @tparam U Type of rangePrefixInfo class which is a container for the data needed to define a key range
+   */
+  private[hbase] abstract class KeyStrategy[T, U] {
+    def getKey(rowKeyInfo: T): Array[Byte]
+    def getKeyRangePrefix(rangePrefixInfo: U): (Array[Byte], Array[Byte])
+  }
+
+  private[hbase] case class KeyStrategy1rowKeyInfo(contigName: String,
```

Comment:
Can you explain the name here? I don't quite get it. Class looks good otherwise though.